pkg/upgrade/upgrade.go

```diff
@@ -21,6 +21,9 @@ func NewUpgradeArgsFunction(flagSource bool) func(cmd *cobra.Command, argv []str
 
 		if flagSource {
 			flags := cmd.Flags()
+			if !cmd.Flags().Changed(machinepoolFlagName) {
```

Comment:
@Mischulee Could you add to our existing unit test to verify that if the flag has not been changed, we do not perform the validation.

cheers!

Rob

Reply:
Added!